memiavl/db.go

```diff
@@ -203,9 +203,57 @@ func Load(dir string, opts Options) (*DB, error) {
 		}
 	}
 
+	if err := removeAllTmpDirs(db.dir); err != nil {
+		db.logger.Error("failed to remove tmp directories", "err", err)
+	}
+
 	return db, nil
 }
 
+func removeAllTmpDirs(rootDir string) error {
+	const maxRemovals = 1000
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+	var errors []error
+	var i int
+
+	dirEntries, err := os.ReadDir(rootDir)
+	if err != nil {
+		return err
+	}
+
+	for _, dirEntry := range dirEntries {
+		if !dirEntry.IsDir() || !strings.HasSuffix(dirEntry.Name(), "-tmp") {
+			continue
+		}
+
+		if i >= maxRemovals {
+			break
+		}
+		i++
+
+		wg.Add(1)
+		go func(dirEntry os.DirEntry) {
```

Comment:
Is `RemoveAll` so slow that need concurrency to optimize it?

Reply:
Just a limit since I'm not sure how many total removal to handle with

Reply:
if it's trivial then don't need the extra code complication, and concurrency has it's own overhead.

Reply:
Can I simplify this code?

Reply:
done, just think concurrent is better in for loop